Attempt to invoke virtual method Article.getTitle() on a null object reference

reported through Google developer console



Android-Version
    Android 6.01

Gerät
    Nexus 5 (hammerhead)

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{fr.gaulupeau.apps.InThePoche/fr.gaulupeau.apps.Poche.ui.ReadArticleActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String fr.gaulupeau.apps.Poche.entity.Article.getTitle()' on a null object reference
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2416)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476)
	at android.app.ActivityThread.-wrap11(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1344)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:148)
	at android.app.ActivityThread.main(ActivityThread.java:5417)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String fr.gaulupeau.apps.Poche.entity.Article.getTitle()' on a null object reference
	at fr.gaulupeau.apps.Poche.ui.ReadArticleActivity.onCreate(ReadArticleActivity.java:116)
	at android.app.Activity.performCreate(Activity.java:6251)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1107)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2369)
	... 9 more
```

--- a/app/src/main/java/fr/gaulupeau/apps/Poche/ui/ReadArticleActivity.java
+++ b/app/src/main/java/fr/gaulupeau/apps/Poche/ui/ReadArticleActivity.java
@@ -101,6 +101,7 @@ public class ReadArticleActivity extends BaseActionBarActivity {
 
         Intent intent = getIntent();
         long articleId = intent.getLongExtra(EXTRA_ID, -1);
+        Log.d(TAG, "onCreate() articleId=" + articleId);
         if(intent.hasExtra(EXTRA_LIST_FAVORITES)) {
             contextFavorites = intent.getBooleanExtra(EXTRA_LIST_FAVORITES, false);
         }
@@ -109,10 +110,28 @@ public class ReadArticleActivity extends BaseActionBarActivity {
         }
 
         DaoSession session = DbConnection.getSession();
+        if(session == null) {
+            Log.e(TAG, "onCreate() DaoSession is null. Thus we" +
+                    " are not able to create this activity. Finish.");
+            finish();
+            return;
+        }
         mArticleDao = session.getArticleDao();
+        if(mArticleDao == null) {
+            Log.e(TAG, "onCreate() ArticleDao is null. Thus we" +
+                    " are not able to create this activity. Finish.");
+            finish();
+            return;
+        }
         mArticle = mArticleDao.queryBuilder()
                 .where(ArticleDao.Properties.Id.eq(articleId)).build().unique();
 
+        if(mArticle == null) {
+            Log.e(TAG, "onCreate() Did not find article with articleId=" + articleId + ". Thus we" +
+                    " are not able to create this activity. Finish.");
+            finish();
+            return;
+        }
         titleText = mArticle.getTitle();
         Log.d(TAG, "onCreate: titleText=" + titleText);
         originalUrlText = mArticle.getUrl();

--- a/app/src/main/java/fr/gaulupeau/apps/Poche/ui/ReadArticleActivity.java
+++ b/app/src/main/java/fr/gaulupeau/apps/Poche/ui/ReadArticleActivity.java
@@ -110,19 +110,7 @@ public class ReadArticleActivity extends BaseActionBarActivity {
         }
 
         DaoSession session = DbConnection.getSession();
-        if(session == null) {
-            Log.e(TAG, "onCreate() DaoSession is null. Thus we" +
-                    " are not able to create this activity. Finish.");
-            finish();
-            return;
-        }
         mArticleDao = session.getArticleDao();
-        if(mArticleDao == null) {
-            Log.e(TAG, "onCreate() ArticleDao is null. Thus we" +
-                    " are not able to create this activity. Finish.");
-            finish();
-            return;
-        }
         mArticle = mArticleDao.queryBuilder()
                 .where(ArticleDao.Properties.Id.eq(articleId)).build().unique();
 

